### PR TITLE
Compiler warning fix and use getNominalL3Field function

### PR DESCRIPTION
**Pull request by victor-gonzalez**

### CODE REVIEW

File: Common/TableProducer/trackextension.cxx [hunk 1]
```diff
@@ -58,7 +58,7 @@ struct TrackExtension {
   Service<o2::ccdb::BasicCCDBManager> ccdb;
 
   int mRunNumber;
-  float mMagField;
+  int mMagField;
```
Comment (ktf) [inline Common/TableProducer/trackextension.cxx:61]: If you are going to convert the mag field to float in any case, I am not sure this is a good idea.

Comment (victor-gonzalez) [reply Common/TableProducer/trackextension.cxx:61]: You are right!
We will get it as an `int` but if we store it as such we will need the conversion on a per particle basis!
I will store it as a float then
Thanks!!

**Comment by jgrosseo** [comment]
I propose to change mMagField to int to make it consistent

**Review by jgrosseo — APPROVED** [review]

**Comment by jgrosseo** [comment]
@victor-gonzalez I propose to move the initializing of the members out of init and directly where you declare the variables.

**Pull request merged**